### PR TITLE
[#8] Implement `MonadMeasured` effect

### LINKED ISSUE #8: Implement `MonadMeasured` effect
Only with EKG is enough I guess...

### PR DESCRIPTION
**Pull request by vrom911**
Resolves #8

### CODE REVIEW

File: src/CakeSlayer/Measure.hs [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+in the @timings@ distribution with the label of the action.
+
+This is a recommended implementation which has the integration with @prometheus@.
+-}
+timedActionImpl
+    :: forall r m a .
+       ( MonadReader r m
+       , Has Timings r
+       , Has Metrics.Store r
+       , Has PrometheusRegistry r
```
Comment (vrom911) [inline src/CakeSlayer/Measure.hs]: I'm not sure, what is the best here: do it with such constraints (like it is right now) or just use it as function arguments, so the one who would use it in the app should do all the `grab` etc. stuff in the instance implementation?

Comment (chshersh) [reply src/CakeSlayer/Measure.hs]: @vrom911 I think that doing it with constraints is fine and in line with our architecture. But we need to think, how to refactor and separate this function into two, so users can choose whatever they want...

File: src/CakeSlayer/Measure.hs [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+{- | This type class is used to perform action and measure time and other
+metrics needed to be performed.
+-}
+class Monad m => MonadTimed m where
+    timedAction :: HasCallStack => m a -> m a
+
+
+-- QUESTION: move to CakeSlayer.Env ?
+type Timings = IORef (Map Text Distribution)
+type PrometheusRegistry = IORef (HashMap Text Histogram)
```
Comment (chshersh) [inline src/CakeSlayer/Measure.hs]: I think it's fine to have them here. Btw, what do you think about converting them to `newtype`s?

Comment (vrom911) [reply src/CakeSlayer/Measure.hs]: I'm not sure. What are the benefits? I think that for these particular types it is not necessary.

Comment (chshersh) [reply src/CakeSlayer/Measure.hs]: @vrom911 Okay, I agree. Maybe if in future there will be overlap or some problem with instances, we can convert to `newtype`s. For now, this is ok :+1:

**Pull request merged**